File: app/src/main/java/org/hackillinois/android/common/DirectionsOnClickListener.java
package org.hackillinois.android.common;

import android.content.Intent;
import android.net.Uri;
import android.view.View;

import com.google.android.gms.maps.model.LatLng;

public class DirectionsOnClickListener implements View.OnClickListener {
    private static final String GOOGLE_MAPS_URI = "http://maps.google.com/maps?q=loc:%s,%s (%s)";

    private LatLng location;
    private String name;

    public DirectionsOnClickListener(LatLng location, String name) {
        this.location = location;
        this.name = name;
    }

    @Override
    public void onClick(View v) {
        String uri = String.format(GOOGLE_MAPS_URI, location.latitude, location.longitude, name);
        Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(uri));
        v.getContext().startActivity(intent);
    }
}

File: app/src/main/java/org/hackillinois/android/common/DirectionsOnClickListener.kt
package org.hackillinois.android.common

import android.content.Intent
import android.net.Uri
import android.view.View
import com.google.android.gms.maps.model.LatLng

class DirectionsOnClickListener(
    private val location: LatLng,
    private val name: String
) : View.OnClickListener {
    private val URI_TEMPLATE = "http://maps.google.com/maps?q=loc:%s,%s (%s)"

    override fun onClick(view: View) {
        val uri = String.format(URI_TEMPLATE, location.latitude, location.longitude, name)
        val intent = Intent(Intent.ACTION_VIEW, Uri.parse(uri))
        view.context.startActivity(intent)
    }
}